discovery/spec.md

```diff
@@ -917,108 +931,111 @@ Content-Type: application/json
 
 Other responses are allowed, but not defined by this specification.
 
-The format of the HTTP Request MUST adhere to the following:
+The format of the HTTP request MUST adhere to the following:
 ```
 Content-Type: application/json
 
 {
   "url": "{url}",
-  "epoch": "{id}",
   "name": "{name}",
   ... remainder of Service attributes ...
 }
 ```
 
-The format of the `200 OK` HTTP Response MUST adhere to the following:
+The format of the `200 OK` HTTP response MUST adhere to the following:
 ```
 200 OK
 Content-Type: application/json
 
 {
   "id": "{id}",
-  "epoch": "{id}",
+  "epoch": "{epoch}",
   ... remainder of Service attributes ...
 }
 ```
 
-#### `DELETE /services/{id}`
+#### `DELETE /services/{id}[?epoch={epoch}]`
+
+This MUST delete the Service as identified by the `id` in the URL. If there
+is no Service with the specified `id` then it is considered to have already
+been deleted and the request will have no change to the Discovery Endpoint.
+A missing Service is not considered an error condition.
 
-This MUST delete the Service at the referenced URL.
+This specification does not define a Body for the request message and the
+presence of a Body (even with a Service definition, valid or not) MUST NOT
+result in an error.
 
-The request MAY include a body. Any such body MUST be an object containing an
-`id` matching the `{id}` in the path. That object MAY include an `epoch`
-value. If the Service corresponding with the specified `id` has a higher
-recorded `epoch` value than given, the request MUST fail.
+The request URL MAY include an OPTIONAL `epoch` query parameter, and if
+present and is not greater than the Service's current `epoch` value, then the
+request MUST fail.
 
-If other service attributes are included in the request, those MUST be
+If other service attributes are included in the request, those SHOULD be
 ignored for the purposes of processing the request.
 
-The follow responses are defined by this specification:
-- `200 OK` if the new Service was deleted.
-  - The HTTP Response Body MUST include the Service values at the time of
-    deletion.
-- `400 Bad Request` if `id` is in the body and does not match the `{id}` in the
-  path
-- `404 Not Found` if there is no Service with the specified `id`
-- `409 Conflict` if the given epoch was less than or equal to the existing
-  epoch of the Service with the given `id`
+Upon successfully processing the request, the HTTP response body MUST be
+a Service definition that includes at least the `id` attribute and SHOULD
+include the remaining Service attributes as they existed immediately prior
+to the Service being deleted, if possible. If the Service was deleted prior
+to the processing of this request the Service value might not be available
+to be returned.
 
-Other responses are allowed, but not defined by this specification.
+If the response Services include an `epoch` value then each MUST be greater
```

Comment:
There's only a single response Service in this case, so I think these lines need a bit of rewording.

Reply:
FIxed